src/appimagetool.c

```diff
@@ -903,6 +908,33 @@ main (int argc, char *argv[])
                         printf("Will not guess update information since zsyncmake is missing\n");
                     }
                 }
+            } else if (github_repository != NULL && github_ref != NULL) {
+                printf("Running on GitHub Actions\n");
+                gchar *zsyncmake_path = g_find_program_in_path ("zsyncmake");
+                if (zsyncmake_path) {
+                    if (strstr(github_ref, "/pull/")) {
+                        printf("Will not calculate update information for GitHub because this is a pull request\n");
+                    } else {
+                        printf("Guessing update information based on $GITHUB_REPOSITORY=%s and $GITHUB_REF=%s\n", github_repository, github_ref);
+                        char buf[1024];
+                        gchar **parts = g_strsplit (github_repository, "/", 2);
+                        char* channel;
+                        if (github_ref != "" && strstr(github_ref, "/continuous/")) {
+                            channel = "latest";
+                        } else {
+                            channel = "continuous";
+                        }
+                        int is_zsync_write_success = snprintf(buf, 1024, "gh-releases-zsync|%s|%s|%s|%s*-%s.AppImage.zsync", parts[0], parts[1], channel, app_name_for_filename, arch);
+                        if (is_zsync_write_success < 0) {
+                            printf("Writing updateinformation failed. zsync information is too long. (> 1024)\n");
```

Comment:
Misleading. You're not writing to the file. You're trying to compose a string.

The numeric value should not be hardcoded in there. Use a const variable.